intranet/apps/eighth/models.py

```diff
@@ -928,6 +933,10 @@ def add_user(self, user, request=None, force=False, no_after_deadline=False):
                 acts = EighthActivity.restricted_activities_available_to_user(user)
                 if self.activity.id not in acts:
                     exception.Restricted = True
+            else:
```

Comment:
I'm not seeing that blacklisting is disabled for restricted activities. And even if it's unlikely, we might as well support the restricted+blacklisted case :P